docker/node/Dockerfile

```diff
@@ -1,13 +1,37 @@
 FROM noaagsl/slurm-node:latest
 
-RUN apt-get update -y && apt-get install -y \
-    libpmi2-0-dev
+#Set environment variables for installing tzdata
+ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+ENV CC=gcc
+ENV CXX=g++
+ENV FC=gfortran
 
-# Set up Intel and Chiltepin environment initialization
-USER admin
-RUN echo "source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1" >> /home/admin/.bashrc
-RUN echo "source /etc/profile.d/conda_init.sh > /dev/null 2>&1" >> /home/admin/.bashrc
-RUN echo "conda activate chiltepin" >> /home/admin/.bashrc
-RUN echo "source /opt/intel/oneapi/setvars.sh > /dev/null 2>&1" >> /home/admin/.bash_profile
-RUN echo "source /etc/profile.d/conda_init.sh > /dev/null 2>&1" >> /home/admin/.bash_profile
-RUN echo "conda activate chiltepin" >> /home/admin/.bash_profile
+RUN apt-get -yqq update && apt-get -yqq upgrade && apt-get -yqq install \
+    bc \
+    build-essential \
+    ca-certificates \
+    cpp \
+    curl \
+    file \
+    g++ \
+    gcc \
+    gfortran \
+    git \
+    git-lfs \
+    gnupg2 \
+    gpg \
+    iproute2 \
+    libpmi2-0-dev \
+    libqt5svg5-dev \
+    locales \
+    make \
+    python3 \
+    python3-pip \
+    python3-setuptools \
```

Comment:
I'm seeing a repetitive pattern with this code here that is also contained in `/master/Dockerfile` and `/frontend/Dockerfile`. I'm assuming this is needed due to the different services it is calling? Just wanted to double check that you were able to DRY this out as much as possible.

Reply:
This file is mostly generated by a spack-stack command, and so most of its contents and formatting is a result of the auto-generation process.  However, I need to make a small manual change to it, so I've added the slightly modified version here.  The only part that is not original is the `specs:` section.  Given that this Dockerfile must be generated with spack-stack commands, it is not productive to waste time reformatting it.  Every time spack-stack is updated and we have a new container, we'd have to make all those changes again. It might be better to completely automate the process entirely so that we create this Dockerfile on-the-fly with a script rather than saving a copy of it.

Reply:
Dockerfiles are not like programs.  They have no "include" mechanism. So there really isn't any good way to dry them out.